### PR TITLE
use immutable collections in api descriptions

### PR DESCRIPTION
**Pull request by t3hnar**

### CODE REVIEW

File: .sbtopts [hunk 1]
```diff
@@ -1,2 +1,3 @@
 -J-Xmx2G
+-J-Xms2G
```
[the remark this answers is not in the document]
Comment (t3hnar) [reply .sbtopts]: the thing is that when you have system wide override of `xms` + `xmx`, startup of sbt with `xmx` override only which is lower than your's `xms` results in fatal error and not very clear message. Hence I propose to always override both `xms` and `xmx`, as JVM validates them against each other. But I'm personally not strong on this line. This just improves user experience and in the end value is not that high

File: core/shared/src/main/scala/zio/Fiber.scala [hunk 2]
```diff
@@ -556,23 +557,21 @@ object Fiber {
   final val dump: UIO[Iterable[Dump]] = UIO.effectSuspendTotal {
     import internal.FiberContext
 
-    def loop(fibers: Iterable[FiberContext[_, _]], acc: UIO[Vector[Dump]]): UIO[Vector[Dump]] =
+    def loop(fibers: List[FiberContext[_, _]], acc: UIO[Vector[Dump]]): UIO[Vector[Dump]] =
```
[the remark this answers is not in the document]
Comment (t3hnar) [reply core/shared/src/main/scala/zio/Fiber.scala]: that's fair comment in general, however there are already few `maps`, `collect`, `flatten` in scope, so performance wise affect should not be visible. And truly saying 
I replaced/improved 
```
val children   = collected1.map(_._1).flatten
           val dumps      = collected1.map(_._2)
```
with `unzip` call, and took that performance improvements back by calling `toList` :D

Other solution would be to not rely on `ZIO.collectAll`, which is the one enforcing this.

File: .sbtopts (hunk 1 above)
[the remark this answers is not in the document]
Comment (vasilmkd) [reply .sbtopts]: I've never had this happen. Ok. I'll let the others decide.

File: core/shared/src/main/scala/zio/stm/TMap.scala [hunk 3]
```diff
@@ -215,7 +217,7 @@ object TMap {
   /**
    * Makes a new `TMap` that is initialized with specified values.
    */
-  final def make[K, V](data: (K, V)*): STM[Nothing, TMap[K, V]] = fromIterable(data)
+  final def make[K, V](a: (K, V), as: (K, V)*): STM[Nothing, TMap[K, V]] = fromIterable(a :: as.toList)
```
[the remark this answers is not in the document]
Comment (t3hnar) [reply core/shared/src/main/scala/zio/stm/TMap.scala]: sure, will do

**Comment by jdegoes** [comment]
I think this is a great pull request, but we should be careful to not perform additional conversions, but rather, always use `immutable.Iterable` wherever we were using `mutable.Iterable`. 

In some cases, for usability or performance reasons, we may wish to support `mutable.Iterable`, but we could make such cases take a by-name parameter (`=> mutable.Iterable[A]`) to indicate we _expect_ the results to be potentially non-deterministic.

Unfortunately ZIO has changed in the weeks since this has been opened and the PR has bit-rotted a bit. I will close for now but feel free to re-open at a convenient time.

Thank you for your work on this! 🙏

**Pull request closed without merging**